From b36f802243c1df8443baf5e0d4c93654a16a9542 Mon Sep 17 00:00:00 2001
From: thebaron <thebaron06@gmail.com>
Date: Thu, 25 Oct 2018 19:45:34 +0200
Subject: [PATCH] Use /run instead of /tmp for resolv.conf.auto

This commit is based on the work of Parthiban Nallathambi. It only
corrects some lines.
Following is the original commit message.

---

From: Parthiban Nallathambi <pn@denx.de>
Date: Fri, 8 Jun 2018 18:41:43 +0200
Subject: [PATCH] [resolv.conf.auto]: Use /run instead of /tmp

Using netifd as network manager along with systemd provokes the bad usage
of path, in this case /tmp/. All the runtime configuration needs to be
sotored as part of /run

Problem:
========
assuming: ln -fs /tmp/resolv.conf.auto /etc/resolv.conf

If any application tend to use Filesystem namespace to have separate /tmp and
try to use DNS resolution from netifd, it will not be able to resolve the path/symlink.

Usage of PrivateTmp=yes:
------------------------
For example, systemd-timesyncd.service in systemd uses the option "PrivateTmp=yes"
in the service file to have separate /tmp filesystem namespace and uses glibc
getaddrinfo for name resolution, which in turm follows the symlink to /tmp which
doesn't exit in new namespace

So moving the runtime configuration to /run here

Signed-off-by: Parthiban Nallathambi <pn@denx.de>

---

Signed-off-by: thebaron <thebaron06@gmail.com>
---
 netifd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/netifd.h b/netifd.h
index 5a90858..e4b6938 100644
--- a/netifd.h
+++ b/netifd.h
@@ -37,7 +37,7 @@
 #define DEFAULT_MAIN_PATH	"/lib/netifd"
 #define DEFAULT_CONFIG_PATH	NULL /* use the default set in libuci */
 #define DEFAULT_HOTPLUG_PATH	"/sbin/hotplug-call"
-#define DEFAULT_RESOLV_CONF	"/tmp/resolv.conf.auto"
+#define DEFAULT_RESOLV_CONF	"/run/resolv.conf.auto"
 #endif
 
 extern const char *resolv_conf;
-- 
2.17.1

